batavia/vendor/VENDORING.md

```diff
@@ -0,0 +1,69 @@
+External JavaScript Dependencies
+================================
+
+External JavaScript dependencies live in the `batavia.vendored` namespace. The
+`batavia/vendor/requirements.js` file contains a list of the currently available
+dependencies. Batavia developers consuming already vendored dependencies don't
+need to know anything else or install extra tools.
+
+The code for the vendored dependencies is bundled in the
```

Comment:
Hi @glasnt, thanks for the quick replies.

The rationale behind committing node_modules to repo is that @freakboy3742 requested that coupling to external code sources be kept to a minimum, and I didn't argue, because there are good reasons for that. 

Access to npm repos (or to github, should we decide to use source dependencies instead of packaged ones) should be only required for the action of downloading and committing the code for each dependency. Not for building the project.

Given a batavia git repo, anyone should be able to work without installing node & other JS tooling. Batavia prides itself in being a welcoming, accessible project, and complexity of tooling would be a limit to that, and not worth it in the absence of other benefits.

Also, with the proposed JS vendoring system, one can download the Batavia repo, go offline, switch branches around, and nothing that previously work will stop working because you went offline. If Make depended on npm/yarn to download JS dependencies on demand, you could `git branch checkout <foo>` and find yourself unable to work because crucial pieces are missing.

In the future, we will probably want to move the whole Batavia build over to a Node.js based tool, and gain the advantages of source maps, dead code elimination, etc. Giles Bowkett persuaded me of this with this article of his about [letting the ruby on rails asset pipeline die](http://gilesbowkett.blogspot.com.au/2016/10/let-asset-pipeline-die.html). If we think of it, Batavia is not a Python project, but a JS one with some Python in it, same as CPython is primarily a C project with some Python in it. I know that I'm mostly writing JS, with a tiny bit of Python when I fiddle with tests.

If/when I learn how to work with JS build systems better and work the above source maps etc magic, I will propose a node-based build tool to replace npm. Then the benefits will outweigh the complexity disadvanages. However, even in that case I'd still advocate for keeping the node_modules directory versioned in-repo, for the stated reasons.